src/overlays/actors/ovl_En_Weiyer/z_en_weiyer.c

```diff
@@ -52,59 +63,590 @@ static ColliderCylinderInit D_80B33A90 = {
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weiyer/EnWeiyer_Draw.s")
+void EnWeiyer_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnWeiyer* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    ActorShape_Init(&this->actor.shape, 1000.0f, ActorShadow_DrawCircle, 65.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06003410, &D_060012AC, this->jointTable, this->morphTable, 19);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->actionFunc = func_80B32804;
+}
+
+void EnWeiyer_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnWeiyer* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80B32384(EnWeiyer* this) {
+    this->unk_196 = this->actor.shape.rot.y;
+    this->unk_27C = (cosf(-M_PI / 8) * 3.0f) + this->actor.world.pos.y;
+    Animation_MorphToLoop(&this->skelAnime, &D_06000FC0, -5.0f);
+    this->unk_194 = 30;
+    this->actor.speedXZ = CLAMP_MAX(this->actor.speedXZ, 2.5f);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->unk_280 = this->actor.floorHeight;
+    this->actionFunc = func_80B328E8;
+}
+
+void func_80B32434(EnWeiyer* this) {
+    Animation_MorphToLoop(&this->skelAnime, &D_06000FC0, -5.0f);
+    this->collider.base.atFlags |= AT_ON;
+    this->unk_194 = 0;
+    this->actor.speedXZ = 5.0f;
+    this->actionFunc = func_80B32C2C;
+}
+
+void func_80B32494(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000288, 2.0f, 0.0f, 0.0f, 0, -8.0f);
+    this->unk_194 = 40;
+    this->collider.base.atFlags |= AT_ON;
+    this->actionFunc = func_80B32D30;
+}
+
+void func_80B32508(EnWeiyer* this) {
+    this->unk_194 = 200;
+    this->collider.base.atFlags |= AT_ON;
+    this->skelAnime.playSpeed = 3.0f;
+    this->actionFunc = func_80B32E34;
+}
+
+void func_80B32538(EnWeiyer* this) {
+    this->unk_194 = 200;
+    this->unk_196 = this->actor.yawTowardsPlayer + 0x8000;
+    this->unk_27C = this->actor.world.pos.y;
+    this->actor.speedXZ = CLAMP_MAX(this->actor.speedXZ, 4.0f);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->skelAnime.playSpeed = 1.0f;
+    this->actionFunc = func_80B33018;
+}
+
+void func_80B325A0(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000FC0, 2.0f, 0.0f, 0.0f, 0, -3.0f);
+    this->unk_194 = 40;
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.speedXZ = 3.0f;
+    func_8003426C(&this->actor, 0x4000, 0xC8, 0, 0x28);
+    this->collider.dim.height = sCylinderInit.dim.height;
+    this->actionFunc = func_80B331CC;
+}
+
+void func_80B32660(EnWeiyer* this) {
+    Animation_Change(&this->skelAnime, &D_06000288, 2.0f, 0.0f, 0.0f, 0, -8.0f);
```

Comment:
in all of the `Animation_Change`s in this file, you can use the animation mode enum for `mode`